### PR TITLE
[SYCL] fix multi-gpu issue on sycl

### PR DESCRIPTION
**Pull request by ClarkChin08**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [ ] Low
  - [ ] Medium
  - [x] High

fix multi-gpu crash issue by filter the platforms of sycl.

### CODE REVIEW

**Comment by airMeng** [comment]
@luoyu-intel

**Comment by airMeng** [comment]
@ClarkChin08 can you attach the measurements results? like llama3-70B on 8 GPUs, memory consumption on each GPU, performance?

File: ggml/src/ggml-sycl/dpct/helper.hpp [hunk 1]
```diff
@@ -2459,6 +2522,7 @@ namespace dpct
                 batch_size);
             break;
         }
+#endif
```
Comment (OuadiElfarouki) [inline ggml/src/ggml-sycl/dpct/helper.hpp:2525]: Why are we reverting the type_combinations for non `__INTEL_MKL__` (namely Nvidia Targets) here ? - this was introduced in https://github.com/ggerganov/llama.cpp/pull/8236 and non Intel targets shall support these `gemm_batch` type-configurations.

File: ggml/src/ggml-sycl/dpct/helper.hpp [hunk 2]
```diff
@@ -2624,6 +2687,7 @@ namespace dpct
                 beta, c, ldc, stride_c, batch_size);
             break;
         }
+#endif
```
Comment (OuadiElfarouki) [inline ggml/src/ggml-sycl/dpct/helper.hpp:2690]: Same as above, the `#ifdef __INTEL_MKL__` block should only cover the first two type-combinations that are not supported by non intel targets.

File: ggml/src/ggml-sycl/dpct/helper.hpp [hunk 3]
```diff
@@ -855,15 +860,69 @@ namespace dpct
...
+            char* env = getenv("ONEAPI_DEVICE_SELECTOR");
+            if (env) {
+                if (std::strstr(env, "level_zero")) {
+                    filter = "level-zero";
+                }
+                else if (std::strstr(env, "opencl")) {
+                    filter = "opencl";
+                }
+                else {
+                    throw std::runtime_error("invalid device filter: " + std::string(env));
```
Comment (OuadiElfarouki) [inline ggml/src/ggml-sycl/dpct/helper.hpp:887]: The filtering approach here excludes non-intel devices and thus breaks the `GGML_SYCL_TARGET==NVIDIA` path in particular.

Comment (NeoZhangJianyu) [reply ggml/src/ggml-sycl/dpct/helper.hpp:887]: rm "filter" won't impact the result and also support non-intel GPU.

**Review by luoyu-intel — APPROVED** [review]

**Review by OuadiElfarouki — APPROVED** [review]
All good thank you!

**Review by airMeng — APPROVED** [review]

**Pull request merged**